uitools/common/src/AuthenticatorController.cpp

```diff
@@ -269,35 +270,18 @@ void AuthenticatorController::continueWithUsernamePasswordArcGIS_(const QString&
                                password,
                                std::nullopt).then(this, [this](TokenCredential* credential)
   {
-    if (m_arcGISPreviousFailureCountsForUrl.contains(m_currentArcGISChallenge->requestUrl()))
-    {
-      m_arcGISPreviousFailureCountsForUrl.remove(m_currentArcGISChallenge->requestUrl());
-    }
-
     m_currentArcGISChallenge->continueWithCredential(credential);
     m_currentArcGISChallenge.reset();
   }).onFailed(this, [this](const ErrorException& e)
   {
-    const auto requestUrl = m_currentArcGISChallenge->requestUrl();
-    if (m_arcGISPreviousFailureCountsForUrl.contains(requestUrl))
+    if (!m_currentArcGISChallenge)
     {
-      m_arcGISPreviousFailureCountsForUrl[requestUrl] = m_arcGISPreviousFailureCountsForUrl[requestUrl] + 1;
-    }
-    else
-    {
-      m_arcGISPreviousFailureCountsForUrl[requestUrl] = 1;
-    }
-
-    if (m_arcGISPreviousFailureCountsForUrl[requestUrl] >= s_maxArcGISPreviousFailureCount)
-    {
-      m_arcGISPreviousFailureCountsForUrl.remove(requestUrl);
-      m_currentArcGISChallenge->continueAndFailWithError(e.error());
-      m_currentArcGISChallenge.reset();
       return;
     }
-
     emit previousFailureCountChanged();
+    m_currentArcGISChallenge->continueWithError(e.error());
     emit displayUsernamePasswordSignInView();
```

Comment:
I don't know if this is correct anymore. Previously we kept track of when to display this dialog with one challenge, but now more challenges will be issued if this is needed. Now we should just call `continueWithError` and let additional challenges happen, or not happen as appropriate.